atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -368,6 +374,102 @@ void WebFrame::ClearCache(v8::Isolate* isolate) {
     base::MemoryPressureListener::MEMORY_PRESSURE_LEVEL_CRITICAL);
 }
 
+v8::Local<v8::Value> WebFrame::Opener() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Opener()->ToWebLocalFrame();
```

Comment:
`ToWebLocalFrame` would [cause assertion](https://cs.chromium.org/chromium/src/third_party/WebKit/Source/core/exported/WebRemoteFrameImpl.cpp?type=cs&sq=package:chromium&l=93) under Debug mode when called for remote web frame, we should call `IsWebLocalFrame()` before doing conversion.

All the `ToWebLocalFrame` calls bellow should also be checked.

Reply:
I would like to understand when a WebFrame is local vs remote.
Does it depend on the chosen process model? 
What's the current model?  Process per window (which would mean all frames are local)?
Is the process model going to change (so that not all frames are local)?
Will it be possible to configure the process model?

And now to the details, what should these methods return if the frame is not local?

Reply:
Regardless of process model, it is always possible to have a remote WebFrame inside any page , especially after Chromium started to enable out-of-process iframes.

I think for now it is fine to return null for remote WebFrame.